### ISSUE TITLE
Remove Search from room options if not available

### ISSUE THREAD

**Issue by daniellekirkwood** [vector-im/element-android#4641]
### Problem
When a user clicks the overflow menu, finds Search, and clicks it, they are told it is not available. This is frustrating

### Solution
If an action (like search) is not available because of something _we did (or don't do)_ then it should not be viewable by users. Remove the Search button if we do not provide search. 

**Note**: If Search is unavailable due to the room type or configuration set by the user then Search should be available from the list (as this is the users limitation, not ours).

**Comment by mnaturel** [comment]
Hi @daniellekirkwood , I tried to reproduce this issue and investigate the corresponding source code. The only scenario I found in which the search is said to be not available is when the room is encrypted (see the capture below). I guess we are in the case of room configuration meaning the search should still be available in the list if I understood well.

![dialog-issue-4641](https://user-images.githubusercontent.com/46314705/152133143-4ae482d0-b8da-449f-bceb-79543f42c86d.png)

Just to be sure: we only have this issue in a room screen, right ? If yes, should we still do something ?
